code/drasil-gool/lib/GOOL/Drasil/State.hs

```diff
@@ -548,7 +553,33 @@ getWordIndex = gets (^. (contentsIndices . _2))
 resetIndices :: MethodState -> MethodState
 resetIndices = set contentsIndices (0,0)
 
+useVarName :: String -> MethodState -> MethodState
+useVarName v = over (varNames . at prefix) (Just . max nextSuffix . fromMaybe 0)
+  where (prefix, nextSuffix) = over _2 (maybe 0 (+1)) $ splitVarName v
+
+genVarName :: [String] -> String -> MS String
```

Comment:
It would be good to have some documentation about the sequence of variable names generated.